src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1588,10 +1610,22 @@ void start(ThreadContainer container) {
     public void run() {
         Runnable task = holder.task;
         if (task != null) {
-            task.run();
+            Object bindings = scopedValueBindings();
+            runWith(bindings, task);
         }
     }
 
+    @Hidden
+    @ForceInline
+    // The VM recognizes this method as special, so any changes to the
+    // name or signature require corresponding changes in
+    // JVM_FindScopedValueBindings().
```

Comment:
Minor nit but I'd prefer to keep things consistent with the existing code/style where we can. In this case, we can move the comment to move the annotations with the /* .. */ comments so it's the same as the other methods.